cpg_infra/github_wif/driver.py

```diff
@@ -0,0 +1,436 @@
+            raise ValueError(f'Project {project_id} missing "repositories"')
+
+
+def sanitize_sa_name(
+    repo_name: str, environment: str, max_length: int = SA_NAME_MAX_LENGTH
+) -> str:
+    """
+    Create a service account name from repo and environment, handling length limits.
+
+    GCP service account IDs must be:
+    - 6-30 characters
+    - Lowercase letters, digits, hyphens
+    - Start with lowercase letter
+
+    Args:
+        repo_name: GitHub repository name
+        environment: Environment name (e.g., 'development', 'production')
+        max_length: Maximum length for service account name
+
+    Returns:
+        Sanitized service account name
+    """
+    # Remove any non-alphanumeric characters and convert to lowercase
+    clean_repo = re.sub(r'[^a-z0-9-]', '-', repo_name.lower())
+    clean_env = re.sub(r'[^a-z0-9-]', '-', environment.lower())
+
+    # Shorten environment name if needed
+    env_short = clean_env
+    if clean_env == 'development':
+        env_short = 'dev'
+    elif clean_env == 'production':
+        env_short = 'prod'
+
+    # Try full name first: {repo}-img-{env}-deployer
+    sa_name = f'{clean_repo}-img-{env_short}-deployer'
+
+    # If too long, truncate repo name
+    if len(sa_name) > max_length:
+        # Pattern: {repo}-img-{env}-deployer
+        # Fixed parts: '-img-' (5 chars) + '-deployer' (9 chars) = 14 chars
+        # Variable part: env_short
+        fixed_overhead = len('-img-') + len('-deployer')  # 14
+        available_for_repo = max_length - len(env_short) - fixed_overhead
+        clean_repo = clean_repo[:available_for_repo]
+        sa_name = f'{clean_repo}-img-{env_short}-deployer'
+
+    # Ensure it starts with a letter
+    if sa_name[0].isdigit():
+        sa_name = 'gh-' + sa_name[3:]  # Replace first 3 chars with 'gh-'
+
+    return sa_name
+
+
+def get_wif_provider_resource_name(project_number: str) -> pulumi.Output[str]:
+    """
+    Get the full resource name for the WIF provider.
+
+    Args:
+        project_number: GCP project number
+
+    Returns:
+        Full WIF provider resource name
+    """
+    return pulumi.Output.concat(
+        f'projects/{project_number}/locations/global/',
+        f'workloadIdentityPools/{WIF_POOL_NAME}/',
+        f'providers/{WIF_PROVIDER_NAME}',
+    )
+
+
+def check_or_create_wif_pool(project_id: str) -> gcp.iam.WorkloadIdentityPool | None:
+    """
+    Check if WIF pool exists, create if it doesn't.
+
+    Args:
+        project_id: GCP project ID
+
+    Returns:
+        WorkloadIdentityPool resource or None if it already exists
+    """
+    return gcp.iam.WorkloadIdentityPool(
+        f'{project_id}-github-pool',
+        workload_identity_pool_id=WIF_POOL_NAME,
+        project=project_id,
+        display_name='GitHub Actions Pool',
+        description='Workload Identity Pool for GitHub Actions OIDC',
+        disabled=False,
+        opts=pulumi.ResourceOptions(
+            # Don't fail if it already exists
```

Comment:
I'm not sure if this is accurate? I can't see anything in the docs indicating that protect=True will prevent failure if it already exists?
https://www.pulumi.com/docs/iac/concepts/resources/options/protect/